From e1f8726cdf403fcf326a70f174e31e3411bde261 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 03 Dec 2023 11:09:10 +0200
Subject: [PATCH] [FSW-334] makefile: move emb specs path to dev makefile

It was decided to separate the FS1 emb specs headers and start from
a clean slate. because of that using 1 include path to emb specs
location for all devices is not possible anymore. the include path is
moved to device specific makefiles.

Change-Id: I9ae5fa5b8163ade310453306bf673e54aae37a9a
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/Makefile b/Makefile
index b5cc2e7..fb53d86 100644
--- a/Makefile
+++ b/Makefile
@@ -12,8 +12,6 @@
 
 CFLAGS += -I$(SPECS_DIR)
 CFLAGS += -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg
-CFLAGS += -I$(SPECS_EMBEDDED_DIR)
-CFLAGS += -I$(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV)
 CFLAGS += -I ${EMBEDDED_ROOT}/arcboot/arch
 CFLAGS += -Wall -Werror
 CFLAGS += -g -c -fno-builtin
@@ -31,7 +29,6 @@
 endif
 
 CFLAGS += -Os -fdata-sections -ffunction-sections
-CFLAGS += -include $(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV)/$(SPECS_EMB_DEV)_reg_map.h
 SFLAGS := $(CFLAGS) -D__ASSEMBLY__
 
 PPBOOT_LD_SCRIPT := $(HL_DEV_ARCH)/ppboot.lds
diff --git a/arch/gaudi2/common/Makefile b/arch/gaudi2/common/Makefile
index e0291d2..55cb01d 100644
--- a/arch/gaudi2/common/Makefile
+++ b/arch/gaudi2/common/Makefile
@@ -13,3 +13,7 @@
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH_COMMON)/security.c)
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH_COMMON)/tpm2.c)
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH_COMMON)/tpm_auth.c)
+
+CFLAGS += -I$(SPECS_EMBEDDED_DIR)
+CFLAGS += -I$(SPECS_EMBEDDED_DIR)/$(DEV_PREFIX)
+CFLAGS += -include $(SPECS_EMBEDDED_DIR)/$(DEV_PREFIX)/$(DEV_PREFIX)_reg_map.h
diff --git a/arch/gaudi3/common/Makefile b/arch/gaudi3/common/Makefile
index 159d90a..267877c 100644
--- a/arch/gaudi3/common/Makefile
+++ b/arch/gaudi3/common/Makefile
@@ -1,5 +1,8 @@
 CFLAGS += -I$(SPECS_H9_DIR)
 CFLAGS += -I$(SPECS_H9_DIR)/gaudi3/asic_reg
+CFLAGS += -I$(SPECS_EMBEDDED_DIR)
+CFLAGS += -I$(SPECS_EMBEDDED_DIR)/$(DEV_PREFIX)
+CFLAGS += -include $(SPECS_EMBEDDED_DIR)/$(DEV_PREFIX)/$(DEV_PREFIX)_reg_map.h
 
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH_COMMON)/common.c)
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH_COMMON)/$(HL_TARGET).c)
